spotlight/losses.py

```diff
@@ -15,7 +15,42 @@
 from spotlight.torch_utils import assert_no_grad
 
 
-def pointwise_loss(positive_predictions, negative_predictions, mask=None):
+def base_loss(loss, sample_weights=None, mask=None):
```

Comment:
Could we call this `_weighted_loss`? This would reflect the fact that:
1. This is an internal function.
2. Its essence lies in applying weights.

Reply:
I'm not opposed to this name, but it'd be a misnomer in case they call us with a mask and no weights. Although, conceptually at least, a mask could be thought of as a weight of zero.

_base_loss
vs
_weighted_loss
vs
_modified_loss

¯\_(ツ)_/¯ ?

Reply:
If, as you suggest, we subsume masks under weights this will all be fine!